File: Practica1/src/practica1/Bicicleta.java
package practica1;

import java.util.logging.Level;
import java.util.logging.Logger;
import java.util.Random;

public abstract class Bicicleta extends Thread{

    int idBicicleta;
    boolean retirarse;
    
    @Override
    public void run(){
        Random r = new Random();
        System.out.println("(+)Comienza la bicicleta " + idBicicleta);
        try {
            sleep(30000);
        } catch (InterruptedException ex) {
            Logger.getLogger(Bicicleta.class.getName()).log(Level.SEVERE, null, ex);
        }
        if (retirarse){
            System.out.println("(-)La bicicleta " + idBicicleta + " se retira");
        }else{
            try {
                sleep(30000);
            } catch (InterruptedException ex) {
                Logger.getLogger(Bicicleta.class.getName()).log(Level.SEVERE, null, ex);
            }
             System.out.println("(END)La bicicleta " + idBicicleta + " ha llegado");
        }

        
    }
    
    

}

File: Practica1Ruby/pkg/Practica1Ruby-0.0.1/lib/Bicicleta.rb
public
class Bicicleta

	public

	def initialize()
		@idBicicleta
		@retirarse
	end
  
  def correr()
    puts "(+)Comienza la bicicleta #{@idBicicleta}"
    sleep(5)
    if (@retirarse)
      puts "(-)La bicicleta #{@idBicicleta} se retira"
    else
      sleep(5)
      puts "(END)La bicicleta #{@idBicicleta} ha llegado"
    end
  end
  
  def clone()
    
  end
  
end
